bionic/protocols.py

```diff
@@ -156,6 +160,13 @@ def tokenize_file(self, path):
         """Like ``tokenize``, but operates on a serialized value."""
         return tokenization.tokenize(path, read_hashable_bytes_from_file_or_dir)
 
+    def hash_value_type(self, value_type, suppress_warnings):
+        """
+        Converts the type of a value into a unique hash which can be used to
+        invalidate cache.
```

Comment:
nit: "invalidate *the* cache"

But also, I don't think this provides enough context on what's happening here. The key thing seems to me that this gives the protocol an opportunity to add data to the hash which doesn't show up in the serialized output.

Finally, if the framing above makes sense, WDYT about just making this something like just `get_extra_value_hash(self, value, suppress_warnings)` that takes the value itself?